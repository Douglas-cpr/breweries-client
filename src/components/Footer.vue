<script setup lang="ts">
defineProps({
  footerText: {
    type: String,
    default: 'Breweries List Â© 2022',
  }
})
</script>

<template>
  <div class="footer">
    <p class="footer__text">{{ footerText }}</p>
  </div>
</template>

<style scoped>
.footer {
  width: 100%;
  font-weight: bold;
  height: 30px;
  padding: 20px 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: var(--footer-bg);
}
</style>
