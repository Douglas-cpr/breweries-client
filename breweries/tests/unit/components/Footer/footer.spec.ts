
import Footer from '@/components/Footer.vue'
import { shallowMount, VueWrapper } from '@vue/test-utils'

describe('<Footer/>', () => {
  let wrapper: VueWrapper;

  function createComponent(footerText: string | undefined = undefined): void {
    wrapper = shallowMount(Footer, {
      props: {
        footerText
      }
    })
  }

  afterEach(() => {
    wrapper.unmount()
  })

  it('should render correctly', () => {
    createComponent()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render footer with correctly text', () => {
    createComponent('Footer text')

    expect(wrapper.props().footerText).toMatch('Footer text')
  })

  it('should render default text if not pass prop footerText', () => {
    createComponent()
    const defaultFooterText = 'Breweries List Â© 2022'

    expect(wrapper.text()).toEqual(defaultFooterText)
  })
})