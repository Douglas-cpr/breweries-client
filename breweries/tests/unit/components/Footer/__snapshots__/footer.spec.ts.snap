// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should render correctly 1`] = `
"<div class=\\"footer\\">
  <p class=\\"footer__text\\">Breweries List Â© 2022</p>
</div>"
`;
